tornado/routing.py

```diff
@@ -16,73 +16,79 @@
 
 Tornado routes HTTP requests to appropriate handlers using `Router` class implementations.
 
-Any `Router` implementation can be used directly as a ``request_callback`` for
-`~.httpserver.HTTPServer` (this is possible because `Router` implements
-`~.httputil.HTTPServerConnectionDelegate`):
+`Router` interface extends `~.httputil.HTTPServerConnectionDelegate` to provide additional
+routing capabilities. This also means that any `Router` implementation can be used directly
+as a ``request_callback`` for `~.httpserver.HTTPServer` constructor.
+
+`Router` subclass must implement a ``find_handler`` method to provide a suitable
+`~.httputil.HTTPMessageDelegate` instance to handle the request:
 
 .. code-block:: python
 
     class CustomRouter(Router):
         def find_handler(self, request, **kwargs):
-            # some routing logic providing a suitable HTTPMessageDelegate implementation
-            return HTTPMessageDelegateImplementation()
+            # some routing logic providing a suitable HTTPMessageDelegate instance
+            return MessageDelegate(request.connection)
```

Comment:
This is an improvement, since this seems to be a complete example that works. But I think it would be better for readers if we provide an example that's *useful*.

I propose you do what Ben mentioned in the mailing list: "you could route requests to different handlers based on the HTTP verb instead of defining different methods on one class". So let's have:

```
resources = {}

class GetResource(RequestHandler):
    def get(self, path):
        try:
            self.finish(resources[path])
        except KeyError:
            raise HTTPError(404)

class PostResource(RequestHandler):
    def post(self, path):
        resources[path] = self.request.body
```
Then, show us how to route to these request handlers based on request.method. I think this could be a useful pattern for some REST frameworks, and it would help me understand how this all ties together.